stable/firehose/templates/deployment.yaml

```diff
@@ -29,22 +29,35 @@ spec:
         name: {{ .Chart.Name }}
         image: "{{ .Values.firehose.image.repository }}:{{ .Values.firehose.image.tag | default .Chart.AppVersion }}"
         imagePullPolicy: {{ .Values.firehose.image.pullPolicy }}
-        {{- if or (index .Values "init-firehose" "enabled") (gt (len .Values.firehose.volumeMounts) 0) }}
+        {{if or (index .Values "init-firehose" "enabled") (ne (get .Values.secrets "FIREHOSE_SINK_BLOB_GCS_CREDENTIAL_NAME" ) "") (ne (get .Values.secrets "FIREHOSE_DLQ_GCS_CREDENTIAL_NAME" ) "") (ne (get .Values.secrets "FIREHOSE_SINK_BIGQUERY_CREDENTIAL_NAME" ) "")  }}
         volumeMounts:
         {{- if index .Values "init-firehose" "enabled" }}
         - mountPath: /work-dir
           name: workdir
         {{- end }}
-        {{- range $_, $volumeMount := .Values.firehose.volumeMounts }}
-        - name: {{ $volumeMount.name}}
-          mountPath: {{ $volumeMount.mountPath }}
+        {{- if (ne (get .Values.secrets "FIREHOSE_SINK_BLOB_GCS_CREDENTIAL_NAME" ) "") }}
```

Comment:
if we are going to support dynamic secrets from entropy only, then why have these as pre-defined configs? just accept a list of secrets and in the chart just take care of mounting them?

Reply:
sure, but then we have to put the mounting secrets and other secrets (env variables) under different keys (maybe `secrets` and `mountSecrets`). In the current implementation everything comes under `secrets`, so we need to pick specific keys that have mounting secret information.

Reply:
only in the chart definitions. on entropy side, firehose module is responsible for it? it should know which need to be mounted and which need to be exposed as env vars.

Reply:
so are you suggestung that we shall split the two on firehose module under `secrets` and `mountSecrets` ? 
if yes, these list of mount secret keys shall be fixed, for the module to distinguish, right?

Reply:
shall I go ahed with this ^ ?

Reply:
no need to split. in the resource spec. it will still be under `secrets`.

but when doing the actual helm chart generation, firehose module can pick the keys and send it to the right places into the chart.